storage/series/series.go

```diff
@@ -400,11 +400,12 @@ func (s *dbSeries) FetchBlocksMetadata(
 		bufferResults.Close()
 	}
 
-	id := s.opts.IdentifierPool().Clone(s.id)
-
 	res.Sort()
 
-	return block.NewFetchBlocksMetadataResult(id, res), nil
+	// NB(r): Since ID and Tags are garbage collected we can safely
+	// return refs.
```

Comment:
"are garbage collected (never manually finalized)" maybe to be a little more clear

Reply:
I guess this was already an issue, but now that I'm thinking about it, should we add a comment at the server level to not finalize the ID / tags we're returning? I notice you only do it in the shard for things that come from disk but seems like something someone could fuck up easily.

Maybe we should just file an issues to have a wrapper for special types of IDs/tags that does an [invariant violated] or panics when you try to finalize it

Reply:
Unfortunately though that means we can't mix types that are meant to be finalized with ones that aren't meant to be finalized.

We could however return a type that on Finalize does absolutely nothing and is a no-op perhaps.  That could be nice, thoughts?

Reply:
yeah that would be chill. Happy to do it here or file an issue or whatever

Reply:
Yeah I'll file an issue, here it is:
https://github.com/m3db/m3db/issues/611

Reply:
How about just exposing `SetNonFinalizable()` on `ident.ID`? Way less type changing - and a more obvious failure mode - we just panic if someone tries to Finalize() such an ID.

Reply:
The thing is, when we mix IDs we actually want to finalize some and some not (i.e. data coming from disk should be finalized and data coming from in memory shouldn't).

So either the caller has to do `if id.ShouldFinalize() { id.Finalize() }` or you return a type that just doesn't finalize when you call `id.Finalize()`.  However we could definitely do `id.SetFinalizeEnabled(false)` and that would disable calls to `id.Finalize()` doing anything.

Reply:
Yeah so my point was we would only set `NonFinalizable()` on id's in memory, and no user would be expected to call `Finalize` on stuff with that set. They wouldn't have to check `if id.ShouldFinalize() { id.Finalize() }` because we'd panic if they did it. 

More I think about it, I agree exposing a new type is cleaner cause it makes it explicit in the contract.

Reply:
Yeah but then you can't share a `ids []ident.ID` because `ID` implements `Finalize` and the `NonFinalizeable` one won't so it can't be one of the elements in the array.  Also if you make it `ids []ident.NonFinalizeable` then you can't finalize the ones that _do_ need finalizing.

Reply:
I don't believe we share `[]ident.ID` in the code today, we always wrap with some other type - iterator/etc. But yeah adding the additional type becomes a widely cascading change cause pretty much anything that wraps the ident.ID would be affected. 

Maybe the version of `id.SetFinalizeEnabled(false)` turning off `Finalize()` from doing anything would be the lowest cost change?